imports/server/methods/configureEnsureGoogleScript.ts

```diff
@@ -0,0 +1,62 @@
+import { check } from 'meteor/check';
+import { Meteor } from 'meteor/meteor';
+import { Random } from 'meteor/random';
+import MeteorUsers from '../../lib/models/MeteorUsers';
+import Settings from '../../lib/models/Settings';
+import { checkAdmin } from '../../lib/permission_stubs';
+import { SettingType } from '../../lib/schemas/Setting';
+import configureEnsureGoogleScript from '../../methods/configureEnsureGoogleScript';
+import GoogleClient from '../googleClientRefresher';
+import googleScriptContent from '../googleScriptContent';
+
+configureEnsureGoogleScript.define({
+  async run() {
+    check(this.userId, String);
+    checkAdmin(await MeteorUsers.findOneAsync(this.userId));
+
+    const { script } = GoogleClient;
+    if (!script) {
+      throw new Meteor.Error('Google client not ready');
+    }
+
+    let app = await Settings.findOneAsync({ name: 'google.script' });
+    const secret = app?.value.sharedSecret ?? Random.secret();
+    const appContent = await googleScriptContent(secret);
+    if (!app) {
+      const resp = await script.projects.create({
```

Comment:
We should probably do a better job of handling the case where the user has not enabled the Apps Script API before.  Right now, the UI shows only "Saving failed: Internal server error [500]" whereas the server logs tell much more:

```
I20221231-21:53:14.107(-8)? Exception while invoking method 'Setup.methods.ensureGoogleScript' Error: Apps Script API has not been used in project 67591736822 before or it is disabled. Enable it by visiting https://console.developers.google.com/apis/api/script.googleapis.com/overview?project=67591736822 then retry. If you enabled this API recently, wait a few minutes for the action to propagate to our systems and retry.
I20221231-21:53:14.108(-8)?     at Gaxios._request (/Users/zarvox/git/jolly-roger/node_modules/gaxios/build/src/gaxios.js:130:23)
I20221231-21:53:14.108(-8)?     at runMicrotasks (<anonymous>)
I20221231-21:53:14.108(-8)?     at processTicksAndRejections (internal/process/task_queues.js:95:5)
I20221231-21:53:14.108(-8)?  => awaited here:
I20221231-21:53:14.108(-8)?     at Function.Promise.await (/Users/zarvox/.meteor/packages/promise/.0.12.0.1fan3uh.w8gi++os+web.browser+web.browser.legacy+web.cordova/npm/node_modules/meteor-promise/promise_server.js:56:12)
I20221231-21:53:14.108(-8)?     at imports/server/methods/configureEnsureGoogleScript.ts:26:17
I20221231-21:53:14.108(-8)?     at /Users/zarvox/.meteor/packages/promise/.0.12.0.1fan3uh.w8gi++os+web.browser+web.browser.legacy+web.cordova/npm/node_modules/meteor-promise/fiber_pool.js:43:40
I20221231-21:53:14.108(-8)?  => awaited here:
I20221231-21:53:14.108(-8)?     at Function.Promise.await (/Users/zarvox/.meteor/packages/promise/.0.12.0.1fan3uh.w8gi++os+web.browser+web.browser.legacy+web.cordova/npm/node_modules/meteor-promise/promise_server.js:56:12)
I20221231-21:53:14.108(-8)?     at getCurrentMethodInvocationResult (packages/ddp-server/livedata_server.js:781:32)
I20221231-21:53:14.108(-8)?     at Meteor.EnvironmentVariable.EVp.withValue (packages/meteor.js:1259:12)
I20221231-21:53:14.108(-8)?     at packages/ddp-server/livedata_server.js:791:46
I20221231-21:53:14.108(-8)?     at new Promise (<anonymous>)
I20221231-21:53:14.109(-8)?     at Session.method (packages/ddp-server/livedata_server.js:739:23)
I20221231-21:53:14.109(-8)?     at packages/ddp-server/livedata_server.js:603:43
```

Since the specific instructions come from Google, perhaps it makes sense to propagate any errors for that particular call?  Alternately, we can give more instructions about enabling the API, and make it a 3-substep setup flow.

Reply:
Done, for both. I added instructions about enabling the relevant APIs (turns out you have to do it in two different places, thanks Google), and also propagated the exception text from that error.

I also made one more ergonomics tweak to the process - I added a generic "ping" RPC method that we now use in `configureGoogleScriptUrl` to make sure that Jolly Roger can actually talk the URL you provide. I think this is most useful at detecting if you accidentally grab the "test" deployment URL, which requires authentication.

In the process of testing that change, I also realized that "updating" the source code of the project didn't actually work. Deployments are pegged to a specific source code revision, so updating the project contents doesn't automatically update any existing deployments. Now it does.

I left all of that in a separate commit (96abfb7) for ease of review, but I'll rebase it into 1840b47 before merging